File: rolemadelen/120812_최빈값_구하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> array) {
    int arr[1001] = {0};
    for(int x : array) ++arr[x];
    
    int max = 0;
    int pos = -1;
    bool multipleExists = false;
    
    for(int i=0; i<1001; ++i) {
        if(arr[i] == max) {
            multipleExists = true; 
        } else if(arr[i] > max) {
            max = arr[i]; 
            pos = i;
            multipleExists = false;
        }
    }
    
    return (multipleExists) ? -1 : pos;
}

File: gamsojji/programmers/beginner/120812_최빈값_구하기.js
function solution(array) {
    var answer =[] , max = 0, idx = 0;
    
    array.forEach(item => {
        answer[item] == null? answer[item] = 1: answer[item] +=1;
    });
    answer.forEach((item, index)=> {
        if(item != 0 && max < item){
            max = item;
            idx = index;
        }else if(item == max){
            idx = -1;
            return;
        }
    })
    
    return idx;
}